src/test/java/tests/UpdateDownloadLinkTest.java

```diff
@@ -0,0 +1,196 @@
+package tests;
+
+import org.junit.Test;
+import updater.UpdateDownloadLink;
+import util.Version;
+
+import java.lang.reflect.Field;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.fail;
+
+public class UpdateDownloadLinkTest {
+
+    private static final String MALFORMED_URL_MESSAGE = "URL is not well formed.";
+
+    @Test
+    public void updateDownloadLinkCompareTo_sameVersionDiffLocation_sameObject() {
+        Version version = new Version(1, 0, 0);
+        UpdateDownloadLink a = new UpdateDownloadLink();
+        a.setVersion(version);
+        try {
+            a.setApplicationFileLocation(new URL("http://google.com"));
+        } catch (MalformedURLException e) {
```

Comment:
These can be `throws` clauses, since the URL is set by us. They'll make the test fail.